lib/Doctrine/Common/EventManager.php

```diff
@@ -68,7 +70,7 @@ public function dispatchEvent($eventName, EventArgs $eventArgs = null)
      *
      * @return array The event listeners for the specified event, or all event listeners.
      */
-    public function getListeners($event = null)
+    public function getListeners(?string $event = null): array
```

Comment:
You had me at "consistence", but the rest is a very good explanation. Thank you for taking the time to clarify this rule of thumb.